scripts/linkbot/monitor-ub-server.js

```diff
@@ -28,27 +28,41 @@ JSDOM.fromURL(url, {}).then(dom => {
         .filter(ref => ref.textContent.match(reg))
         .filter(ref => !ref.textContent.match(reg).groups.minor) // Ignore the minor number
         .reduce((acc, ref) => {
-            acc.push(ref.textContent.match(reg))
-            return acc
+            acc.push(ref.textContent.match(reg));
+            return acc;
         }, [])
         .sort((a,b) => {
-            return parseInt(b.groups.release) - parseInt(a.groups.release)
-        })
-    var matchOn = matches[0].input
+            return parseInt(b.groups.release) - parseInt(a.groups.release);
+        });
+    var matchOn = matches[0].input;
 
     return matchOn;
 
     // Get the release image url from the releases (sub)-page
     // const url = "http://cdimage.ubuntu.com/ubuntu/releases/"
 }).then(releaseVersion => {
-    var releaseVersion = releaseVersion.replace(/\s/g, "").replace(/\//g, "")
-    JSDOM.fromURL(`${url}${releaseVersion}/release/`, {}).then(dom => {
+    var releasedVersion = releaseVersion.replace(/\s/g, "").replace(/\//g, "");
+    JSDOM.fromURL(`${url}${releasedVersion}/release/`, {}).then(dom => {
         var document = dom.window.document;
         var refs = document.getElementsByTagName("a");
-        const match = Array.from(refs).find(ref => ref.href.match(`.*ubuntu-${releaseVersion}\.?[0-9]+-preinstalled-server-armhf.*\.img\.xz$`))
+        const match = Array.from(refs).find(ref => ref.href.match(`.*ubuntu-${releasedVersion}\.?[0-9]+-preinstalled-server-armhf.*\.img\.xz$`));
         if (match) {
-            console.log(`Ubuntu server image has a new release: ${match}`)
-            updateURLLink(`${target}=\"${match}\"`, target)
+            console.log(`Ubuntu server image has a new release: ${match}`);
+            updateURLLink(`${target}=\"${match}\"`, target);
         }
     })
-});
+        .catch(err => {
+            console.log('Failed to get the update URL');
+            console.log(`${url}${releasedVersion}/release/`);
+            JSDOM.fromURL(`${url}${releasedVersion}/beta/`, {})
+                .then(dom => {
+                    console.log('Only the beta is out still');
+                });
+        });
+})
+    .catch(err => {
+        console.log('Failed to get the URL');
+        console.log(url);
+        console.log(err);
+        throw err;
+    });
```

Comment:
This should work, but it's not very pretty IMO... you could return each `JSDOM.fromURL` call to get rid of the nesting and help readability, but I'm fine with merging it as is... "additional changes are left as a reader exercise" or something along those lines 😄